stats/bridge_stats.go

```diff
@@ -181,20 +181,38 @@ func (s *Service) Start() {
 				errTimer.Reset(10 * time.Second)
 				continue
 			}
-			stopChn := make(chan struct{}, 1)
-			// process report loop
-			go func() {
-				s.reportLoop(conn)
-				stopChn <- struct{}{}
-			}()
-
-			// process read loop
-			go func() {
-				s.readLoop(conn)
-				stopChn <- struct{}{}
-			}()
-			<-stopChn
-			errTimer = time.NewTimer(0)
+			go s.readLoop(conn)
+
+			if err = s.report(conn); err != nil {
+				log.Warn("Initial stats report failed", "err", err)
+				conn.Close()
+				errTimer.Reset(0)
+				continue
+			}
+
+			sendStatsTicker := time.NewTicker(5 * time.Second) // Every 5 seconds
+			for err == nil {
+				select {
+				case <-s.quitCh:
+					sendStatsTicker.Stop()
+					conn.Close()
+					return
+				case msg := <-s.errCh:
+					err = s.setLastError(msg.Listener, msg.Err)
+				case msg := <-s.processedBlockCh:
+					err = s.setProcessedBlock(msg.Listener, msg.ProcessedBlock)
+				case <-sendStatsTicker.C: // Checking stats interval
+					if err = s.report(conn); err != nil {
+						log.Warn("bridge stats report failed", "err", err)
+						// When bridge stats report failed need to relogn after 10 seconds
+					}
+				}
+			}
```

Comment:
In my opinion, this looks unclean with nested for-select, I think we should move this to a separate function.

Reply:
agree, I think we should put reportLoop function back

Reply:
ah just want to make a consistent flow with ethstats, so in the future, in case we want to send stats from the other we can reuse, but it senses let me update it